src/revocation.rs

```diff
@@ -10,48 +10,143 @@ use crate::error::*;
 use crate::secure_mount;
 
 use std::convert::TryInto;
-use std::io::Write;
+use std::fs;
+use std::io::{ErrorKind, Write};
 use std::path::{Path, PathBuf};
-use std::process::{Command, Output, Stdio};
+use std::process::{Child, Command, Output, Stdio};
 
 use serde_json::Value;
 
+/// Lookup for the action to be executed and return the command string
+///
+/// The lookup goes in the following order:
+/// 1. Look for pre-installed action
+/// 2. Look for the action in the tenant-provided initial payload
+/// 3. Look for pre-installed Python action
+/// 4. Look for the Python action in the tenant-provided initial payload
+fn lookup_action(
+    payload_dir: &Path,
+    actions_dir: &Path,
+    action: &str,
+    allow_payload_actions: bool,
+) -> Result<(String, bool, bool)> {
+    let mut py_action = PathBuf::from(action);
+    if !py_action.set_extension("py") {
+        return Err(Error::Other(format!(
+            "unable to set action {} extension",
+            &action
+        )));
+    }
+
+    // The tuple is considered as (script, is_python, is_payload)
```

Comment:
```suggestion
    // The tuple is considered as (script, is_python, is_payload)
    // This creates four possible paths that will be searched to see if the script exists. The order corresponds to the lookup order described in the documentation for this function.
```
Just because it took me a minute to wrap my head around this.

Reply:
Will add.